Removes the check which restricts CMake build types.

This version of the patch applies to LLVM starting from revision
90dcdc4b6e7d86cb3d5049bd766aecddd549dd7d.

--- a/llvm/CMakeLists.txt	2015-08-12 20:04:55.397881480 +0200
+++ b/llvm/CMakeLists.txt	2015-08-12 20:05:06.625741113 +0200
@@ -340,10 +340,6 @@
 
 string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
 
-if (CMAKE_BUILD_TYPE AND
-    NOT uppercase_CMAKE_BUILD_TYPE MATCHES "^(DEBUG|RELEASE|RELWITHDEBINFO|MINSIZEREL)$")
-  message(FATAL_ERROR "Invalid value for CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
-endif()
 
 # LLVM_INSTALL_PACKAGE_DIR needs to be declared prior to adding the tools
 # subdirectory in order to have the value available for llvm-config.
